docs/user-guide/02-conventional-features.md

```diff
@@ -792,3 +792,394 @@ In this example, `globalUniform` is appended to the global root signature / pipe
+    * We generate a 3 different visibilities of 'member-wise constructor' in order:
+        1. `public` 'member-wise constructor'
+            * Contains members of visibility: `public`
+            * Do not generate if `internal` or `private` member lacks an init expression
+        2. `internal` 'member-wise constructor'
+            * Contains members of visibility: `internal`, `public`
+            * Do not generate if `private` member lacks an init expression
+        3. `private` 'member-wise constructor'
+            * Contains members of visibility: `private`, `internal`, `public`
+```C#
+struct GenerateCtorInner1
+{
+    internal int a = 0;
+    
+    /// Implicitly generate:
+    // internal __init(int in_a)
+    // {
+    //     a = 0;
+    //
+    //     a = in_a;
+    // }
+};
+struct GenerateCtor1 : GenerateCtorInner1
+{
+    internal int b = 0;
+    public int c;
+
+    /// Implicitly generate:
+    // internal __init(int in_a, int in_b, int in_c)
+    // {
+    //     b = 0;
+    //
+    //     this = GenerateCtorInner1(in_a);
+    //
+    //     b = in_b;
+    //     c = in_c;
+    // }
+    //
+    // public __init(int in_c)
+    // {
+    //     b = 0;
+    //
+    //     this = GenerateCtorInner1();
+    //
+    //     c = in_c;
+    // }
+};
+
+struct GenerateCtorInner2
+{
+    internal int a;
+    /// Implicitly generate:
+    // internal __init(int in_a)
+    // {
+    //     a = in_a;
+    // }
+};
+struct GenerateCtor2 : GenerateCtorInner2
+{
+    internal int b;
+    public int c;
+
+    /// Note: `internal b` is missing init expression,
+    // Do not generate a `public` 'member-wise' constructor.
+
+    /// Implicitly generate:
+    // internal __init(int in_a, int in_b, int in_c)
+    // {
+    //     this = GenerateCtorInner2(in_a);
+    //
+    //     b = in_b;
+    //     c = in_c;
+    // }
+};
+```
+
+Initializer Lists
+----------
+Initializer List's are an expression of the form `{...}`.
+
+This form is **different** from declaring a new scope
+```c#
+int myFunc()
+{
+    { // New Scope
+        int a = {}; // Initializer List
+    }
+}
+```
+
+### Initializer List's - Scalar
+
+```c#
+// Equivalent to `int a = 1`
+int a = {1};
+```
+
+### Initializer List's - Vectors
+
+```c#
+// Equivalent to `float3 a = float3(1,2,3)`
+float3 a = {1, 2, 3};
+```
+
+### Initializer List's - Arrays/Matrixes
+
+#### Array Of Scalar's
+
+```c#
+// Equivalent to `int[2] a; a[0] = 1; a[1] = 2;`
+int a[2] = {1, 2}
+```
+
+#### Array Of Aggregate's
+
+```c#
+// Equivlent to `float3 a[2]; a[0] = {1,2,3}; b[1] = {4,5,6};`
+float3 a[2] = {{1,2,3}, {4,5,6}};
+```
+#### C-Style-Initializer-List
+
+```c#
+// Equivalent to `float3 a[2] = {{1,2,3}, {4,5,6}};`
+float3 a[3] = {1,2,3, 4,5,6}; 
+```
+
+### Initializer Lists - Struct
+
+Slang has 2 seperate calling style for an Initializer List with `struct`'s:
```

Comment:
" In most scenarios, using an initializer list to create a struct typed value is equivalent to calling the struct's constructor using the elements in the initilaizer list as arguments". For example: 
...
"